fastapi_pagination/ext/beanie.py

```diff
@@ -30,15 +30,27 @@ async def paginate(
     lazy_parse: bool = False,
     **pymongo_kwargs: Any,
 ) -> Any:
-    params, raw_params = verify_params(params, "limit-offset")
+    params, raw_params = verify_params(params, "limit-offset", "cursor")
+    if additional_data is None:
+        additional_data = {}
 
     if isinstance(query, AggregationQuery):
         aggregation_query = query.clone()  # type: ignore
         paginate_data = []
-        if raw_params.limit is not None:
-            paginate_data.append({"$limit": raw_params.limit + (raw_params.offset or 0)})
-        if raw_params.offset is not None:
-            paginate_data.append({"$skip": raw_params.offset})
+        if is_limit_offset(raw_params):
+            if raw_params.limit is not None:
+                paginate_data.append({"$limit": raw_params.limit + (raw_params.offset or 0)})
+            if raw_params.offset is not None:
+                paginate_data.append({"$skip": raw_params.offset})
+        else:
+            if raw_params.cursor:  # type: ignore[attr-defined]
+                paginate_data.append(
+                    {
+                        "_id": {  # type: ignore[dict-item]
+                            "$gt": PydanticObjectId(raw_params.cursor),  # type: ignore[attr-defined]
```

Comment:
Can it raise an exception? Can we add validation that `cursor` is valid `PydanticObjectId`?